contracts/rbn-staking/FeeCustody.sol

```diff
@@ -0,0 +1,425 @@
+
+  // Intermediary path asset for univ3 swaps.
+  // Empty if direct pool swap between asset and distribution asset
+  mapping(address => bytes) public intermediaryPath;
+
+  // Oracle between asset/usd pair for total
+  // reward approximation across all assets earned
+  mapping(address => address) public oracles;
+
+  address[1000] public assets;
+  // Index of empty slot in assets array
+  uint256 public lastAssetIdx;
+
+  // Keeper for weekly distributions
+  address public keeper;
+
+  // Events
+  event NewAsset(address asset, bytes intermediaryPath);
+  event RecoveredAsset(address asset);
+  event NewFeeDistributor(address feeDistributor);
+  event NewRBNLockerAllocation(uint256 pctAllocationForRBNLockers);
+  event NewDistributionToken(address distributionToken);
+  event NewProtocolRevenueRecipient(address protocolRevenueRecipient);
+  event NewKeeper(address keeper);
+
+  /**
+   * @notice
+   * Constructor
+   * @param _pctAllocationForRBNLockers percent allocated for RBN lockers (100% = 10000)
+   * @param _feeDistributor address of fee distributor where protocol revenue claimable
+   * @param _protocolRevenueRecipient address of multisig
+   * @param _admin admin
+   * @param _keeper keeper
+   */
+  constructor(
+    uint256 _pctAllocationForRBNLockers,
+    address _feeDistributor,
+    address _protocolRevenueRecipient,
+    address _admin,
+    address _keeper
+  ) {
+    require(_feeDistributor != address(0), "!_feeDistributor");
+    require(
+      _protocolRevenueRecipient != address(0),
+      "!_protocolRevenueRecipient"
+    );
+    require(_admin != address(0), "!_admin");
+    require(_keeper != address(0), "!_keeper");
+
+    pctAllocationForRBNLockers = _pctAllocationForRBNLockers;
+    feeDistributor = IFeeDistributor(_feeDistributor);
+    protocolRevenueRecipient = _protocolRevenueRecipient;
+    keeper = _keeper;
+  }
+
+  /**
+   * @notice
+   * Swaps RBN locker allocation of protocol revenu to distributionToken,
+   * sends the rest to the multisig
+   * @dev Can be called by keeper
+   * @param _minAmountOut min amount out for every asset type swap.
+   * will need to be in order of assets in assets[] array. should be
+   * fine if we keep track.
+   * @return toDistribute amount of distributionToken distributed to fee distributor
+   */
+  function distributeProtocolRevenue(uint256[] calldata _minAmountOut)
+    external
+    returns (uint256 toDistribute)
+  {
+    require(msg.sender == keeper, "!keeper");
+
+    if (address(distributionToken) == WETH) {
+      IWETH(address(distributionToken)).deposit{value: address(this).balance}();
+    }
+
+    for (uint256 i; i < lastAssetIdx; i++) {
+      IERC20 asset = IERC20(assets[i]);
+      uint256 assetBalance = asset.balanceOf(address(this));
+
+      if (assetBalance == 0) {
+        continue;
+      }
+
+      uint256 multiSigRevenue = assetBalance
+        .mul(TOTAL_PCT.sub(pctAllocationForRBNLockers))
```

Comment:
We can remove SafeMath because Solidity already supports overflow/underflow checks. So we can remove all instances of `mul` and `sub`

Reply:
Agree since 0.8 not needed.